### PR TITLE
Impliment support for categories attribute

### PR DESCRIPTION
**Pull request by mhdzumair**
Moodle create icallender file with categories attribute. I highly need this attribute because, with this attribute i can check what is the module code for the event.

### CODE REVIEW

File: test/test_data/categories_test.ics [hunk 1]
```diff
@@ -0,0 +1,27 @@
+BEGIN:VCALENDAR
+METHOD:PUBLISH
+PRODID:-//Moodle Pty Ltd//NONSGML Moodle Version 2019111802//EN
+VERSION:2.0
+BEGIN:VEVENT
+UID:3014@lms.itum.mrt.ac.lk
+SUMMARY:Lecture 2
```
Comment (awawa1) [inline test/test_data/categories_test.ics:7]: Id

File: test/test_data/categories_test.ics [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+DESCRIPTION:
+CLASS:PUBLIC
+LAST-MODIFIED:20201110T094221Z
+DTSTAMP:20201110T095626Z
+DTSTART:20201117T073000Z
```
Comment (awawa1) [inline test/test_data/categories_test.ics:12]: 1

File: test/test_data/categories_test.ics [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+DTEND:20201117T093000Z
+CATEGORIES:In19-S04-IT2403
+END:VEVENT
+BEGIN:VEVENT
+UID:3003@lms.itum.mrt.ac.lk
+SUMMARY:Week 2 (18/11/2020)
+DESCRIPTION:If Password ask you can give the following:Password: it2406-MC
+CLASS:PUBLIC
+LAST-MODIFIED:20201110T062311Z
```
Comment (awawa1) [inline test/test_data/categories_test.ics:21]: Ping

File: test/test_data/categories_test.ics [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+DTSTAMP:20201110T095626Z
+DTSTART:20201118T021500Z
+DTEND:20201118T041500Z
+CATEGORIES:In19-S04-IT2406,In19-S04-IT2405
```
Comment (awawa1) [inline test/test_data/categories_test.ics:25]: Ip

File: test/test_data/categories_test.ics [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+END:VEVENT
+BEGIN:VEVENT
+UID:3003@lms.itum.mrt.ac.lk
+SUMMARY:Week 2 (18/11/2020)
+DESCRIPTION:If Password ask you can give the following:Password: it2406-MC
+CLASS:PUBLIC
+LAST-MODIFIED:20201110T062311Z
+DTSTAMP:20201110T095626Z
+DTSTART:20201118T021500Z
+DTEND:20201118T041500Z
```
Comment (awawa1) [inline test/test_data/categories_test.ics:24]: Start

File: test/test_data/categories_test.ics (hunk 4 above)
Comment (awawa1) [inline test/test_data/categories_test.ics:25]: It2405

File: test/test_icalevents.py [hunk 6]
```diff
@@ -33,14 +33,14 @@ def test_events(self):
...
         ical = "test/test_data/duration.ics"
         start = date(2018, 1, 1)
         end = date(2018, 2, 1)
-        
+
         evs = icalevents.events(file=ical, start=start, end=end)
-        
+
         e1 = evs[0]
         self.assertEqual(e1.start.day, 10, "explicit event start")
```
Comment (awawa1) [inline test/test_icalevents.py:45]: Hotmail

File: test/test_icalevents.py [hunk 7]
```diff
@@ -33,14 +33,14 @@ def test_events(self):
...
         self.assertEqual(e1.end.day, 13, "implicit event end")
```
Comment (awawa1) [inline test/test_icalevents.py:46]: 13

File: icalevents/icalparser.py [hunk 8]
```diff
@@ -47,6 +47,7 @@ def __init__(self):
         self.sequence = None
```
Comment (awawa1) [inline icalevents/icalparser.py:47]: app

File: icalevents/icalparser.py [hunk 9]
```diff
@@ -129,6 +130,7 @@ def copy_to(self, new_start=None, uid=None):
         ne.uid = uid
         ne.created = self.created
```
Comment (awawa1) [inline icalevents/icalparser.py:130]: Ne

File: icalevents/icalparser.py (hunk 8 above)
Comment (awawa1) [inline icalevents/icalparser.py:47]: Self

File: test/test_data/categories_test.ics [hunk 10]
```diff
@@ -0,0 +1,27 @@
+BEGIN:VCALENDAR
+METHOD:PUBLISH
+PRODID:-//Moodle Pty Ltd//NONSGML Moodle Version 2019111802//EN
+VERSION:2.0
+BEGIN:VEVENT
+UID:3014@lms.itum.mrt.ac.lk
+SUMMARY:Lecture 2
+DESCRIPTION:
```
Comment (awawa1) [inline test/test_data/categories_test.ics:8]: Summary

File: test/test_data/categories_test.ics [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+CATEGORIES:In19-S04-IT2403
+END:VEVENT
+BEGIN:VEVENT
+UID:3003@lms.itum.mrt.ac.lk
+SUMMARY:Week 2 (18/11/2020)
+DESCRIPTION:If Password ask you can give the following:Password: it2406-MC
+CLASS:PUBLIC
+LAST-MODIFIED:20201110T062311Z
+DTSTAMP:20201110T095626Z
+DTSTART:20201118T021500Z
```
Comment (awawa1) [inline test/test_data/categories_test.ics:23]: Start

**Review by awawa1 — APPROVED** [review]

**Comment by mhdzumair** [comment]
Hey @awawa1  what are these review?

**Comment by amcappelli** [comment]
Any chance this could be merged soon? I would really like to have this feature as well.

**Comment by mhdzumair** [comment]
> Any chance this could be merged soon? I would really like to have this feature as well.

I'm not sure if it is merged,  
for the moment you can install my forked icalevent by 
`pip install git+https://github.com/mhdzumair/icalevents.git`

or you can also include it to requirements.txt file entry as
 
`git+https://github.com/mhdzumair/icalevents.git#egg=icalevents  ` 

then 

`pip install -r requirements.txt`

**Comment by Hultner** [comment]
> Any chance this could be merged soon? I would really like to have this feature as well.

The project is not actively maintained, I helped a little bit with maintenance since I used the library in the testing stack of a previous project but don’t any more haven’t got much bandwidth over to this. That said I will try to take a look at it next week and merge if it looks good. I’m not merging anything without proper tests as I don’t want the project to destabilize over time but this changed does look like it’s got some tests so  I can hopefully safely merge it.

**Comment by Hultner** [comment]
> Hey @awawa1 what are these review?

Looks like spam to me, deleted the comments.

**Comment by mhdzumair** [comment]
> > Any chance this could be merged soon? I would really like to have this feature as well.
> 
> The project is not actively maintained, I helped a little bit with maintenance since I used the library in the testing stack of a previous project but don’t any more haven’t got much bandwidth over to this. That said I will try to take a look at it next week and merge if it looks good. I’m not merging anything without proper tests as I don’t want the project to destabilize over time but this changed does look like it’s got some tests so I can hopefully safely merge it.

Thanks for consider this PR. I did what I want most. I hope you will be satisfied with my coding cheers!

**Review by Hultner — APPROVED** [review]
Looks great, and there's a proper test for the new feature. Approved :)

**Comment by Hultner** [comment]
@mhdzumair I merged your feature. We'll have to wait for @irgangla to cut a new release before we'll see it on pypi.

**Pull request merged**